Apps/Gateway/src/ClientApp/src/stores/user.ts

```diff
@@ -0,0 +1,468 @@
+        setLoadedStatus(user.value.hdid !== undefined);
+    }
+
+    function handleError(
+        resultError: ResultError,
+        errorType: ErrorType,
+        errorSource: ErrorSourceType
+    ) {
+        logger.error(`Error: ${JSON.stringify(resultError)}`);
+        setUserError(resultError.resultMessage);
+
+        if (resultError.statusCode === 429) {
+            const errorKey = "page";
+            if (errorType === ErrorType.Retrieve) {
+                errorStore.setTooManyRequestsWarning(errorKey);
+            } else {
+                errorStore.setTooManyRequestsError(errorKey);
+            }
+        } else {
+            errorStore.addError(errorType, errorSource, resultError.traceId);
+        }
+    }
+
+    function createProfile(request: CreateUserRequest): Promise<void> {
+        return new Promise((resolve, reject) =>
+            userProfileService
+                .createProfile(request)
+                .then((userProfile) => {
+                    logger.verbose(
+                        `User Profile: ${JSON.stringify(userProfile)}`
+                    );
+                    setProfileUserData(userProfile);
+                    resolve();
+                })
+                .catch((resultError: ResultError) => {
+                    setUserError(resultError.resultMessage);
+                    reject(resultError);
+                })
+        );
+    }
+
+    function retrieveProfile(): Promise<void> {
+        return new Promise((resolve, reject) =>
+            userProfileService
+                .getProfile(user.value.hdid)
+                .then((userProfile) => {
+                    logger.verbose(
+                        `User Profile: ${JSON.stringify(userProfile)}`
+                    );
+                    setProfileUserData(userProfile);
+                    resolve();
+                })
+                .catch((resultError: ResultError) => {
+                    handleError(
+                        resultError,
+                        ErrorType.Retrieve,
+                        ErrorSourceType.Profile
+                    );
+                    reject(resultError);
+                })
+        );
+    }
+
+    function updateAcceptedTerms(termsOfServiceId: string): Promise<void> {
+        return new Promise((resolve, reject) =>
+            userProfileService
+                .updateAcceptedTerms(user.value.hdid, termsOfServiceId)
+                .then((userProfile) => {
+                    logger.verbose(
+                        `User Profile: ${JSON.stringify(userProfile)}`
+                    );
+                    setProfileUserData(userProfile);
+                    resolve();
+                })
+                .catch((resultError: ResultError) => {
+                    handleError(
+                        resultError,
+                        ErrorType.Update,
+                        ErrorSourceType.Profile
+                    );
```

Comment:
```suggestion
                    setUserError(resultError.resultMessage);
```
This used to commit the userError mutation, not the handleError action.